Efforting python3.10 support

akrherz opened the pull request:
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
*  Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Attempting to build off of @dopplershift's work in #58

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

akrherz commented:
Holy cow, I got this PR green on windows.  Trying some more things. cc: @dopplershift

akrherz commented:
Green!  @erogluorhan @michaelavs 

What I did here:

1) build off of dopperlshift's numpy dep cleanup
2) switched to `mingw32` compiler, which worked for me during local testing for unknown reasons :(
3) Merged the windows build steps into one, which worked for me during local testing for unknown reasons :(
4) added myself as a maintainer to help out more going forward
5) applied python 3.10 migration

I have no interest in windows, just wanted to get this green so to move ahead with python 3.10 :)

dopplershift commented on recipe/meta.yaml:
```diff
@@ -21,19 +20,17 @@ requirements:
-    - numpy 1.19.*  # [unix]
-    - numpy 1.14.*  # [win]
+    - numpy
+    # upstream has it as a requirement
```
At *build* time?

akrherz replied:
Otherwise, `requirements.txt` gets pulled in and `wrapt` gets pip installed, which then is evaluated for inclusion in the resulting distribution.  This cuts down build time, iirc

dopplershift replied:
That behavior doesn't even make sense based on upstream's [`setup.py`](https://github.com/NCAR/wrf-python/blob/1.3.2.6/setup.py)? I'm confused.

akrherz replied:
check out [this build log](https://dev.azure.com/conda-forge/84710dde-1620-425b-80d0-4cf5baca359d/_apis/build/builds/443100/logs/67), it is attempting to package wrapt.

dopplershift replied:
That link's not working.

akrherz replied:
darn, it must have gotten culled.  I'll revert that change, so you can see it here.

akrherz replied:
So for example

```
found egg dir: D:\bld\wrf-python_1642543278575\_h_env\Lib\site-packages\wrapt-1.13.3-py3.7-win-amd64.egg
found egg dir: D:\bld\wrf-python_1642543278575\_h_env\Lib\site-packages\wrf_python-1.3.2-py3.7-win-amd64.egg
number of files: 116
```

```
   INFO (wrf-python,Lib/site-packages/wrapt/_wrappers.cp38-win_amd64.pyd): Needed DSO python38.dll found in conda-forge::python-3.8.12-h900ac77_2_cpython
   INFO (wrf-python,Lib/site-packages/wrapt/_wrappers.cp38-win_amd64.pyd): Needed DSO C:/Windows/System32/KERNEL32.dll found in $SYSROOT
   INFO (wrf-python,Lib/site-packages/wrapt/_wrappers.cp38-win_amd64.pyd): Needed DSO C:/Windows/System32/VCRUNTIME140.dll found in $SYSROOT
   INFO (wrf-python,Lib/site-packages/wrapt/_wrappers.cp38-win_amd64.pyd): Needed DSO C:/Windows/System32/downlevel/api-ms-win-crt-runtime-l1-1-0.dll found in $SYSROOT
```

```
ClobberWarning: This transaction has incompatible packages due to a shared path.
  packages: conda-forge/win-64::wrapt-1.13.3-py38h294d835_1, local/win-64::wrf-python-1.3.2.6-py38h16f8063_1
  path: 'lib/site-packages/wrapt/__init__.py'
```

dopplershift replied:
I see it. I don't understand why it's doing it, but I see it... 🤨

dopplershift commented on recipe/bld.bat:
```diff
@@ -6,12 +6,5 @@ cd ..
-
-CALL %PYTHON% setup.py install --single-version-externally-managed --record=record.txt
-
+CALL %PYTHON% setup.py config_fc --f90flags="-O2 -mtune=generic -fopenmp" build_ext --libraries="gomp" build --compiler=mingw32 --fcompiler=gnu95 install
```
Is there a reason `install` is done on this line instead of as:
```
CALL %PYTHON% -m pip install . -vv
```
similar to how it's done in `build.sh` and was (sort of) done before? Just wondering if that has to do with why the "easy install" path is being taken.

akrherz replied:
This was one of the workarounds I found necessary to overcome the failures you found in #58.  I can try it now.

akrherz replied:
I am unsure what to make of the new failure...  Will let this sit before reverting the recent changes.

akrherz commented:
That's interesting to only see python3.10 fail now.  I have no idea why yet :(

akrherz commented:
I have reverted this back to the combination that works on Windows, I realize it is sub-optimal.  I really don't have an appetite to debug windows issues further, but am desperate to get this package going on conda-forge for Linux + python 3.10.

Could @conda-forge/wrf-python please decide what they want to do?

erogluorhan reviewed (approved):
It looks good to me if @dopplershift thinks his points were addressed.

dopplershift reviewed (approved):
Looks good.
